### DISCUSSION TITLE
main thread panic in 1.29.2

### DISCUSSION THREAD

**Discussion by paulens12** [dani-garcia/vaultwarden#3962] [Q&A]
### Subject of the issue
I updated the docker image to 1.29.2 and started getting this error:
```
thread 'main' panicked at 'OS can't spawn worker thread: Operation not permitted (os error 1)', /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.31.0/src/runtime/scheduler/multi_thread/worker.rs:447:13

stack backtrace:
   0:     0x5598dcaa6ca3 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hbd7d55b7108d2ab8
   1:     0x5598dc4ff3df - core::fmt::write::h6d54cd7c9e155ec5
   2:     0x5598dcaa38d7 - std::io::Write::write_fmt::h6a453a71c692f63b
   3:     0x5598dcaa6a75 - std::sys_common::backtrace::print::h4ddf81241a51b337
   4:     0x5598dcaa84b0 - std::panicking::default_hook::{{closure}}::hff91f1f484ade5cd
   5:     0x5598dcaa8274 - std::panicking::default_hook::h21f14afd59f7aef9
   6:     0x5598dcaa8999 - std::panicking::rust_panic_with_hook::h45f66047b14c555c
   7:     0x5598dcaa88a4 - std::panicking::begin_panic_handler::{{closure}}::h49d1a88ef0908eb4
   8:     0x5598dcaa7096 - std::sys_common::backtrace::__rust_end_short_backtrace::hccebf9e57f8cc425
   9:     0x5598dcaa8642 - rust_begin_unwind
  10:     0x5598dc452883 - core::panicking::panic_fmt::h54ec9d0e3180a83d
  11:     0x5598dcae19f4 - tokio::runtime::blocking::pool::Spawner::spawn_blocking::h9e0ded7b77863876
  12:     0x5598dcae0d05 - tokio::runtime::blocking::pool::spawn_blocking::hb15e5178f17f818b
  13:     0x5598dcaf566b - tokio::runtime::scheduler::multi_thread::worker::Launch::launch::h6ce2d7d02be3860d
  14:     0x5598dcb083f9 - tokio::runtime::builder::Builder::build::h6fb748647923ea1f
  15:     0x5598dceff52e - vaultwarden::main::h222f937f9ca88bfc
  16:     0x5598dd2f61d7 - std::sys_common::backtrace::__rust_begin_short_backtrace::h267d3b5af2cab27a
  17:     0x5598dce50c38 - std::rt::lang_start::{{closure}}::h3ae0294076e6f9bb
  18:     0x5598dca9bf15 - std::rt::lang_start_internal::hf502095b101390bb
  19:     0x5598dceff895 - main
  20:     0x7fdfa3dc61ca - <unknown>
  21:     0x7fdfa3dc6285 - __libc_start_main
  22:     0x5598dc4962b1 - _start
  23:                0x0 - <unknown>
```
### Deployment environment

* vaultwarden version: 1.29.2

* Install method: docker-compose

* Clients used: not applicable

* Reverse proxy and version: not applicable, the main container is in a crash loop
* MySQL/MariaDB or PostgreSQL version: none

* Other relevant details:
I'm using a pretty old version of Debian:
`Linux omv 4.19.0-0.bpo.9-amd64 #1 SMP Debian 4.19.118-2+deb10u1~bpo9+1 (2020-06-09) x86_64`
`Docker version 19.03.15, build 99e3ed8919`

### Expected behaviour
Vaultwarden should work

### Actual behaviour
The container exits immediately with the above error. No other information is logged.

**Comment by BlackDex** [comment]
Please update your host os, or at least try to update docker first to the latest docker-ce version available.

**Reply by ghost** [reply]
I have the same issue using Linux 6.1.56-1-lts and docker 24.0.6 build ed223bc820

**Reply by ghost** [reply]
Having looked at the post about the issue on armv6 hosts I tried the alpine based image. That one works. I should point out though that my system is definitely x86/64

**Reply by paulens12** [reply]
that might be tough, my home server setup is quite fragile and I'd rather make a full backup and move things over one-by-one to a new box than risk breaking it by the means of an online update. I will see if I can find time for that anytime soon. In the short term, I think I'll try the Alpine image too.

**Reply by BlackDex** [reply]
The stable version does have some stability issues with Alpine though.
The current testing tagged images shouldn't have this issue.
